iceoryx_utils/test/moduletests/test_concurrent_trigger_queue.cpp

```diff
@@ -14,127 +14,121 @@
-    EXPECT_THAT(this->m_sut->empty(), Eq(false));
-}
-
-TYPED_TEST(TriggerQueue_test, SinglePush)
-{
-    EXPECT_THAT(this->m_sut->push(512u), Eq(true));
-    uint64_t value;
-    EXPECT_THAT(this->m_sut->blocking_pop(value), Eq(true));
-    EXPECT_THAT(value, Eq(512u));
-}
-
-TYPED_TEST(TriggerQueue_test, PushTillItsFull)
-{
-    for (uint64_t i = 0; i < this->m_sut->capacity(); ++i)
-    {
-        ASSERT_THAT(this->m_sut->push(i * 3), Eq(true));
-    }
-
-    for (uint64_t i = 0; i < this->m_sut->capacity(); ++i)
-    {
-        uint64_t value;
-        EXPECT_THAT(this->m_sut->blocking_pop(value), Eq(true));
-        EXPECT_THAT(value, Eq(value));
-    }
+    EXPECT_THAT(this->m_sut.push(5U), Eq(true));
+    EXPECT_THAT(this->m_sut.empty(), Eq(false));
+    EXPECT_THAT(this->m_sut.size(), Eq(1U));
 }
 
-TYPED_TEST(TriggerQueue_test, PushWhenItIsAlreadyFull)
+TYPED_TEST(TriggerQueue_test, PushTillFull)
 {
-    for (uint64_t i = 0; i < this->m_sut->capacity(); ++i)
-    {
-        ASSERT_THAT(this->m_sut->push(i * 3), Eq(true));
-    }
-
-    EXPECT_THAT(this->m_sut->push(551), Eq(false));
+    EXPECT_TRUE(this->fillQueue());
 }
 
-TYPED_TEST(TriggerQueue_test, TryPopOnEmpty)
+TYPED_TEST(TriggerQueue_test, PopOnEmptyReturnsNullopt)
 {
-    uint64_t value;
-    EXPECT_THAT(this->m_sut->try_pop(value), Eq(false));
+    EXPECT_THAT(this->m_sut.pop(), Eq(cxx::nullopt));
 }
 
-TYPED_TEST(TriggerQueue_test, TryPopSingleElement)
+TYPED_TEST(TriggerQueue_test, PopOneElement)
 {
-    uint64_t value;
-    this->m_sut->push(73u);
-    EXPECT_THAT(this->m_sut->try_pop(value), Eq(true));
-    EXPECT_THAT(value, Eq(73u));
+    this->m_sut.push(123U);
+    auto result = this->m_sut.pop();
+    ASSERT_THAT(result.has_value(), Eq(true));
+    EXPECT_THAT(*result, Eq(123U));
 }
 
-TYPED_TEST(TriggerQueue_test, TryPopFull)
+TYPED_TEST(TriggerQueue_test, PopFullQueue)
 {
-    for (uint64_t i = 0; i < this->m_sut->capacity(); ++i)
-    {
-        this->m_sut->push(i * 3u);
-    }
+    constexpr uint64_t OFFSET = 231;
+    this->fillQueue(OFFSET);
 
-    for (uint64_t i = 0; i < this->m_sut->capacity(); ++i)
+    for (uint64_t i = 0U; i < this->m_sut.capacity(); ++i)
     {
-        uint64_t value;
-        EXPECT_THAT(this->m_sut->try_pop(value), Eq(true));
-        EXPECT_THAT(value, Eq(i * 3u));
+        auto result = this->m_sut.pop();
+        ASSERT_THAT(result.has_value(), Eq(true));
+        EXPECT_THAT(*result, Eq(i + OFFSET));
     }
 }
 
-TYPED_TEST(TriggerQueue_test, Size)
+TYPED_TEST(TriggerQueue_test, PushBlocksUntilPopWhenFull)
 {
-    for (uint64_t i = 0; i < this->m_sut->capacity(); ++i)
-    {
-        this->m_sut->push(i * 3);
-        EXPECT_THAT(this->m_sut->size(), Eq(i + 1));
-    }
-}
+    constexpr int64_t TIMEOUT_IN_MS = 100;
+    this->fillQueue();
 
-TYPED_TEST(TriggerQueue_test, Capacity)
-{
-    EXPECT_THAT(this->m_sut->capacity(), helper::GetCapacity(this->m_sut.value()));
+    std::atomic_uint64_t counter{0U};
```

Comment:
On gcc 5.4 is `std::atomic_uint64_t` is not known, you may use instead:
```suggestion
    std::atomic<uint64_t> counter{0U};
```